### PR TITLE
Fix: implicit conversion changes signedness, unused value

### PR DESCRIPTION
**Pull request by jrohel**
Fixes clang compilation errors:
4 x error: implicit conversion changes signedness
1 x error: variable 'total_bytes_read' set but not used

### CODE REVIEW

**Comment by jrohel** [comment]
I have fixed this type of error several times.
Can we re-enable compilation using clang? So it will probably be necessary to include this https://github.com/rpm-software-management/dnf5/pull/1418 to pass the unit tests with clang compiled code.

File: dnf5daemon-client/commands/repoquery/repoquery.cpp [hunk 1]
```diff
@@ -219,16 +219,14 @@ void RepoqueryCommand::run() {
 
     int in_fd = pipefd[0];
 
-    const size_t pipe_buffer_size = fcntl(in_fd, F_GETPIPE_SZ);
+    const auto pipe_buffer_size = static_cast<size_t>(fcntl(in_fd, F_GETPIPE_SZ));
```
Comment (m-blaha) [inline dnf5daemon-client/commands/repoquery/repoquery.cpp:222]: We should handle situation when `fcntl` fails and return -1 and add some reasonable fallback value - 4096?

Comment (jrohel) [reply dnf5daemon-client/commands/repoquery/repoquery.cpp:222]: @m-blaha Good idea. Moreover, F_GETPIPE_SZ is Linux specific, so a fallback would be useful.

Comment (m-blaha) [reply dnf5daemon-client/commands/repoquery/repoquery.cpp:222]: Maybe we can just go with a constant value. Looks like on my system the default value is 65536 and do not try to detect the size.

Comment (m-blaha) [reply dnf5daemon-client/commands/repoquery/repoquery.cpp:222]: Anyway, it's out of this PR scope, I filed a new issue for it - https://github.com/rpm-software-management/dnf5/issues/1419

**Review by m-blaha — APPROVED** [review]
Thank you!

**Comment by m-blaha** [comment]
The test failures are unrelated.

**Pull request merged**